### ISSUE TITLE
ORA-00933 Command not properly ended

### ISSUE THREAD

**Issue by bobf32** [dbeaver/dbeaver#4291]
Hi, 

On issuing a query of the form:

```
SELECT 
    col_a, 
    CASE WHEN col_b = DATE '2018-08-13' THEN 'Up-to-date' ELSE 'Earlier' END col_b,
    count(1)
FROM tab_a l
GROUP BY 
    col_a, 
    CASE WHEN col_b = DATE '2018-08-13' THEN 'Up-to-date' ELSE 'Earlier' END;
```

I get ORA-00933 error when executing SQL. If I highlight everything except the terminating ; the query succeeds. 

DB 5.2.1.

**Comment by masebu** [comment]
Hi,

I found the same issue on 5.2.2 today  also with a query ending in CASE (...) END;
Just upgraded to 5.2.5: seems to be gone, though I was not able to pinpoint an issue or commit precisely about that.

regards

**Comment by masebu** [comment]
Actually, the problem still occurs if I highlight the complete query (including ; ) before executing.
Usually I just have my cursor somewhere in the query or right after the semicolon and I do ctlr+enter: in that case the problem does not occur (it did on 5.2.2).

**Comment by p91paul** [comment]
It was #4248

**Comment by premyslc** [comment]
This error occur when table name contains "end" word and select statements is selected and with ";" at the end. Something like this throw error.

select * from some_end_table;

**Comment by psychotik2k3** [comment]
it still occurs even in 6.3.3 and my table does not have "end" in the name:
`SELECT tsr.col_a, 
CASE WHEN tsr.col_b = 'toto' THEN NULL ELSE 't' END 
FROM TIME_SEGMENT_RESULT tsr;`

**Comment by premyslc** [comment]
May be that it's not depending only on table name. CASE statement cause this error in this case.

**Comment by uslss** [comment]
already fixed in 6.3.5. Ticket closed
